### DISCUSSION TITLE
Building on Raspbery PI

### DISCUSSION THREAD

**Discussion by KHjerrild** [merbanan/rtl_433#2884] [Q&A] [answered]
Hi I'm trying to build the RTL_433 on my raspberry pi. I already have the RTL-SDR running.
I used the build commands described in BUILDING.md
           git clone https://github.com/merbanan/rtl_433.git
           cd rtl_433/
           cmake -B build
Complains that build dir doesn't exist

Trying this one instead:
          mkdir build && cd build && cmake ../ && make

Works fine until CMake Error at CMakeLists.txt:104 (if):

I have attached my log.
Hope someone out there can help me. Thanks in advance.

[build_log.txt](https://github.com/merbanan/rtl_433/files/14733566/build_log.txt)

**Comment by zuckschwerdt** [comment]
Is your cmake a really old version? (`cmake --version`). The compiler version 4.9.2 is 10 years old.
It should create the build dir if needed and it should process those `if` args. Maybe I can find a compatible solution if I can reproduce the environment. Is it some old Raspbian? (`cat /etc/debian_version`)

**Reply by KHjerrild** [reply]
Yeah it's a quite old install.
cmake version : 3.6.2
cat /ect/debian_version : 8.0

**Reply by zuckschwerdt** [reply]
I'll try an install of Debian Jessie and see if we can be compatible still.

**Reply by zuckschwerdt** [reply]
And you can just delete that block, it's only diagnostics. Maybe that was the only blocker.
https://github.com/merbanan/rtl_433/blob/afb5b34709b699bf88aa95b6c577591235a5d591/CMakeLists.txt#L103-L107

**Reply by KHjerrild** [reply]
Thank you for your support. 

It now builds but fails in the linking. I've attached my log. Hope you have yet another trick in your sleeve.
[build_log.txt](https://github.com/merbanan/rtl_433/files/14734936/build_log.txt)

**Reply by zuckschwerdt** [reply, marked as answer]
OpenSSL is found but can't be properly linked. If you don't plan to use TLS (mqtts / influxs) then disable it using
`mkdir build && cd build && cmake -DENABLE_OPENSSL=OFF .. && make`

**Reply by zuckschwerdt** [reply]
There are now hints for building with older CMake (3.13 and 3.6) on https://triq.org/rtl_433/BUILDING.html#cmake
The compatibilty has been fixed with 7751b57

**Reply by KHjerrild** [reply]
Rebased and build with your added instructions. All went without a glitch. Thank you very much.
